File: 3577.count-the-number-of-computer-unlocking-permutations/solution.kt
private const val MOD = 1_000_000_007

class Solution {
    fun countPermutations(complexity: IntArray): Int =
        complexity.drop(1).let { rest ->
            if (rest.any { it <= complexity[0] }) 0 else modFac(rest.size)
        } 
        
    tailrec fun modFac(n: Int, acc: Long = 1): Int =
        if (n == 1) acc.toInt() else modFac(n - 1, n * acc % MOD)
}

File: 3577.count-the-number-of-computer-unlocking-permutations/solution.go
const MOD = 1_000_000_007

func countPermutations(complexity []int) int {
    f := 1
    for i := 1; i < len(complexity); i++ {
        if complexity[i] <= complexity[0] {
            return 0
        }
        f = (f * i) % MOD
    }
    return f
}